src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -767,81 +1158,185 @@ End Get
+End Class", new[]
+{
+    VerifyVB.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.MemberRule).WithLocation(0).WithArguments("DoesNotMatter.PublicP1_"),
+    VerifyVB.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.MemberRule).WithLocation(1).WithArguments("DoesNotMatter.ProtectedP4_"),
+    VerifyVB.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.MemberRule).WithLocation(2).WithArguments("I1.P_"),
+    VerifyVB.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.MemberRule).WithLocation(3).WithArguments("ImplementI1.P2_")
+}, @"
+Public Class DoesNotMatter
+    Public Property PublicP1() As Integer
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+    ' No diagnostic
+    Private Property PrivateP2_() As Integer
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+    ' No diagnostic
+    Friend Property InternalP3_() As Integer
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+    Protected Property ProtectedP4() As Integer
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+End Class
+
+Public Interface I1
+    Property P() As Integer
+End Interface
+
+Public Class ImplementI1
+    Implements I1
+    ' No diagnostic
+    Public Property P() As Integer Implements I1.P
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+    Public Overridable Property P2() As Integer
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+End Class
+
+Public Class Derives
+    Inherits ImplementI1
+    ' No diagnostic
+    Public Overrides Property P2() As Integer
+        Get
+            Return 0
+        End Get
+        Set
+        End Set
+    End Property
+End Class");
         }
 
         [Fact]
         public async Task CA1707_ForEvents_VisualBasic()
         {
-            await VerifyVB.VerifyAnalyzerAsync(@"
+            await VerifyVB.VerifyCodeFixAsync(@"
 Public Class DoesNotMatter
-    Public Event PublicE1_ As System.EventHandler
+    Public Event {|#0:PublicE1_|} As System.EventHandler
     Private Event PrivateE2_ As System.EventHandler
     ' No diagnostic
     Friend Event InternalE3_ As System.EventHandler
     ' No diagnostic
-    Protected Event ProtectedE4_ As System.EventHandler
+    Protected Event {|#1:ProtectedE4_|} As System.EventHandler
 End Class
 
 Public Interface I1
-    Event E_ As System.EventHandler
+    Event {|#2:E_|} As System.EventHandler
 End Interface
 
 Public Class ImplementI1
     Implements I1
     ' No diagnostic
     Public Event E_ As System.EventHandler Implements I1.E_
-    Public Event E2_ As System.EventHandler
+    Public Event {|#3:E2_|} As System.EventHandler
 End Class
 
 Public Class Derives
     Inherits ImplementI1
+
+    Public Shadows Event {|#4:E2_|} As System.EventHandler
```

Comment:
When renaming `E2_` in `Derives` class, the new name "E2" is detected as a clashing, most probably because `E2` already exists in`ImplementI1`. However, we indeed want to rename it. It's marked with Shadows and is meant to shadow the member in `ImplementI1`.

My opinion is that the analyzer don't report for the "Shadows" case, and RenameSymbolAsync gets it updated when https://github.com/dotnet/roslyn/issues/46663 is fixed.

@mavasani @Evangelink What are your opinions?